plasmapy/simulation/particle_tracker.py

```diff
@@ -0,0 +1,715 @@
+        """Save a hdf5 file containing simulation positions and velocities."""
+
+        path = Path(self.output_directory) / f"{particle_tracker.iteration_number}.hdf5"
+
+        with h5py.File(path, "w") as output_file:
+            output_file["x"] = particle_tracker.x
+            output_file["v"] = particle_tracker.v
+
+    def save_to_memory(self, particle_tracker):
+        """Append simulation positions and velocities to save routine object."""
+        self.x_all.append(particle_tracker.x)
+        self.v_all.append(particle_tracker.v)
+
+    @abstractmethod
+    def post_push_hook(self, particle_tracker):
+        """Function called after a push step.
+
+        This function is responsible for handling two steps of save routine, namely:
+            - Deciding whether or not to save on the current time step
+            - How the simulation data is saved (i.e. to disk or memory)
+
+        """
+        ...
+
+
+class DiskSaveRoutine(AbstractSaveRoutine):
+    """Save routine corresponding to saving a hdf5 file every given interval."""
+
+    def __init__(self, output_directory, interval: u.Quantity):
+        super().__init__(output_directory)
+        self.save_interval = interval
+        self.time_of_last_save = 0
+        self.name_of_last_save = 0
+
+    def require_uniform_dt(self):
+        """Save output only makes sense for uniform time steps,
+        therefore this routine requires a uniform time step.
+        """
+        return True
+
+    def save_now(self, particle_tracker):
+        """Save at every interval given in instantiation."""
+        saves_per_step = np.floor(self.save_interval / particle_tracker.dt)
+
+        # If the user is requesting multiple saves per step then raise a ValueError
+        if saves_per_step == 0:
+            raise ValueError(
+                "You must specify a time step smaller than the save interval!"
+            )
+
+        return particle_tracker.time - self.time_of_last_save >= self.save_interval
+
+    def post_push_hook(self, particle_tracker):
+        """Save to disk if one interval has elapsed since last save."""
+
+        if self.save_now(particle_tracker):
+            self.time_of_last_save = particle_tracker.time
+            self.save_to_disk(particle_tracker)
+
+
+class ParticleTracker:
+    """
+    General particle tracer.
+
+    """
+
+    def __init__(
+        self,
+        grids: Union[AbstractGrid, Iterable[AbstractGrid]],
+        req_quantities=None,
+        verbose=True,
+    ):
+        # self.grid is the grid object
+        if isinstance(grids, AbstractGrid):
+            self.grids = [
+                grids,
+            ]
+        elif isinstance(grids, collections.abc.Iterable):
+            self.grids = grids
+        elif isinstance(grids, BasePlasma):
+            raise TypeError(
+                "This class has been deprecated. Please revert to PlasmaPy version 2023.5.1 to use this version of ParticleTracker."
```

Comment:
Maybe "It appears you may be trying to access an older version of the ParticleTracker class. This class has been deprecated..."